lintly/parsers.py

```diff
@@ -199,6 +199,44 @@ def parse_violations(self, output):
         return violations
 
 
+class CfnLintParser(BaseLintParser):
+    """A parser for the `cfn-lint` command.
+
+      cfn-lint output example:
+
+      W2001 Parameter UnusedParameter not used.
+      template.yaml:2:9
+    """
+
+    def parse_violations(self, output):
+        violations = collections.defaultdict(list)
+
+        regex = re.compile(r"[EW]\d{4}\s")
+
+        next_line_is_path = False
+        current_violation = None
+        for line in output.strip().splitlines():
+            if regex.match(line):
+                # This line is a cfn-lint error or warning
+                next_line_is_path = True
+                current_violation = line
+            elif next_line_is_path:
+                # This line is a filepath:line_number:column_number
+                path, line_number, column = line.split(":")
+                path = self._normalize_path(path)
+
+                violation = Violation(line=int(line_number),
+                                      column=int(column),
+                                      code="`cfn-lint`",
```

Comment:
Based on this example:

```
W2001 Parameter UnusedParameter not used.
templates/template.yaml:2:9
```

The `code` should be the first 5 characters (`W2001`) and the `message` should be the rest (`Parameter UnusedParameter not used`). This is how the other linters work as well.

Let's change this up a bit so that the codes are dynamic. Side note, it looks like cfn-lint actually calls them ["rule IDs."]https://github.com/awslabs/cfn-python-lint/blob/master/docs/rules.md)

Reply:
Perfect. The latest changes should take care of that =)